dipy/reconst/mapmri.py

```diff
@@ -230,12 +241,14 @@ def __init__(self,
 
         self.positivity_constraint = positivity_constraint
         if self.positivity_constraint:
-            if not have_cvxopt:
+            if not have_cvxpy:
                 raise ValueError(
-                    'CVXOPT package needed to enforce constraints')
-            if not hasattr(cvxopt, 'solvers'):
-                raise ValueError("CVXOPT version 1.1.7 or higher required")
+                    'CVXPY package needed to enforce constraints')
             msg = "pos_radius must be 'adaptive' or a positive float"
+            if cvxpy_solver is not None:
+                if cvxpy_solver not in cvxpy.installed_solvers():
+                    msg = "cvxpy_solver is not installed in cvxpy."
```

Comment:
It's worth being more explicit here. A useful error is of the form: "I was expecting x, but I got y instead, check z". 

For example, here: 

    msg = "The input `cvxpy_solver` was set to %s."%cvxpy_solver
    msg += "One of %s"% ' , '.join(cvxp.installed_solvers)
    msg += " was expected."